gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/infra/domain_service/analytics_engine/permissions/ApiAnalyticsQueryFilterDecoratorImpl.java

```diff
@@ -44,44 +56,136 @@
 @RequiredArgsConstructor
 public class ApiAnalyticsQueryFilterDecoratorImpl implements AnalyticsQueryFilterDecorator {
 
-    private static final String ENVIRONMENT_ADMIN = RoleScope.ENVIRONMENT.name() + ':' + SystemRole.ADMIN.name();
+    private static final String ORGANIZATION_ADMIN = RoleScope.ORGANIZATION.name() + ':' + SystemRole.ADMIN.name();
 
-    private final ApiAuthorizationService apiAuthorizationService;
+    private static final String UNKNOWN_APPLICATION = "Unknown";
 
-    private final PermissionService permissionService;
+    private final ApiSearchService apiSearchService;
+
+    private final ApplicationService applicationSearchService;
 
     @Override
-    public Map<String, API> getAllowedApis() {
-        var allowedApis = new HashMap<String, API>();
-
-        var executionContext = GraviteeContext.getExecutionContext();
-        if (isEnvironmentAdmin()) {
-            var apiIds = apiAuthorizationService.findIdsByEnvironment(executionContext.getEnvironmentId());
-            apiIds.forEach(id -> allowedApis.put(id, new API("")));
-            return allowedApis;
+    public MetricsContext getFilteredContext(MetricsContext context, List<Filter> requestFilters) {
+        var allowedApis = getAllowedApis(requestFilters);
+        var allowedApplications = getAllowedApplications(requestFilters);
+        var updatedFilters = updateContextFilters(context.filters(), allowedApis);
+
+        return context.withApiNamesById(allowedApis).withApplicationNameById(allowedApplications).withFilters(updatedFilters);
+    }
```

Comment:
To me, the context is not "filtered", it's just a context that we can update from one layer to another.